src/Xenial.Tasty/Remote/NamedPipeRemoteHook.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipes;
+using System.Threading;
+using System.Threading.Tasks;
+
+using StreamJsonRpc;
+
+using Xenial.Delicious.Protocols;
+using Xenial.Delicious.Scopes;
+
+namespace Xenial.Delicious.Remote
+{
+    public delegate Task<bool> IsInteractiveRun();
+
+    public delegate Task<TransportStreamFactory?> TransportStreamFactoryFunctor();
+    public delegate Task<Stream> TransportStreamFactory();
+
+    public delegate Task<TastyRemote> ConnectToRemote(TastyScope scope, Stream remoteStream);
+
+    //[EnumeratorCancellation] 
+    public delegate IAsyncEnumerable<SerializableTestCase> ReportTests(IAsyncEnumerable<SerializableTestCase> tests, CancellationToken cancellationToken);
+
+    public static class TastyRemoteDefaults
+    {
+        public static Task<bool> IsInteractiveRun()
+        {
+            var isInteractive = Environment.GetEnvironmentVariable("TASTY_INTERACTIVE");
+            if (!string.IsNullOrEmpty(isInteractive))
+            {
+                if (bool.TryParse(isInteractive, out var result))
+                {
+                    return Task.FromResult(result);
+                }
+            }
+            return Task.FromResult(false);
+        }
+
+        public static Task<TastyRemote> AttachToStream(TastyScope scope, Stream remoteStream)
+        {
+            var remote = JsonRpc.Attach<TastyRemote>(remoteStream);
+
+            scope.RegisterReporter(test => remote.Report(new SerializableTestCase
```

Comment:
Totally not the way how plugins should inject into remote execution, but without remote reporters, this is a tough one right now

Reply:
We should address this in a future PR